components/content_settings/renderer/brave_content_settings_agent_impl.cc

```diff
@@ -112,6 +121,17 @@ bool BraveContentSettingsAgentImpl::IsReduceLanguageEnabled() {
   return reduce_language_enabled_;
 }
 
+void BraveContentSettingsAgentImpl::BraveSpecificDidAllowJavaScriptOnce(
+    const GURL& resource_url) {
+  // This will be called for all resources on a page, we want to notify only
+  // about frame domain and javascript resources.
```

Comment:
how this can be called for all resources if `IsScriptTemporilyAllowed` checks `temporarily_allowed_scripts_` which should contain only script URLs?

Reply:
it also checks origins so we have here all resources from a page, woff/ttf/json/html